src/js/events/MesosSummaryActions.js

```diff
@@ -104,12 +104,17 @@ var MesosSummaryActions = {
     function (resolve, reject) {
 
       return function (timeScale) {
-        let canAccessHistoryServer = Hooks.applyFilter(
+        let canAccessHistoryAPI = Hooks.applyFilter(
           'hasCapability', false, 'historyServiceAPI'
         );
 
-        if (!_historyServiceOnline || !canAccessHistoryServer) {
-          requestFromMesos(resolve, reject);
+        if (!_historyServiceOnline || !canAccessHistoryAPI) {
+          let canAccessMesosAPI = Hooks.applyFilter(
+            'hasCapability', false, 'mesosAPI'
+          );
+          if (canAccessMesosAPI) {
```

Comment:
Yup, realized this after I put up the PR and that's why now we emit an event with an empty state.
Which turns out to be a very hacky fix. Got any better ideas on how to do this?
The main problem is that a lot of the pages rely on the MesosSummaryStore to say "hey I've received data" and then they render. The real fix is to go through each page and have them rely on their own store saying "hey I've got data".